lib/ui/submission/survey_submission_success_screen.dart

```diff
@@ -0,0 +1,91 @@
+import 'package:flutter/material.dart';
+import 'package:go_router/go_router.dart';
+import 'package:lottie/lottie.dart';
+import 'package:survey_flutter_ic/extension/context_extension.dart';
+import 'package:survey_flutter_ic/theme/dimens.dart';
+
+const _animationDurationInMilliseconds = 2000;
+
+class SurveySubmissionSuccessScreen extends StatefulWidget {
+  const SurveySubmissionSuccessScreen({Key? key}) : super(key: key);
+
+  @override
+  State<SurveySubmissionSuccessScreen> createState() =>
+      _SurveySubmissionSuccessScreenState();
+}
+
+class _SurveySubmissionSuccessScreenState
+    extends State<SurveySubmissionSuccessScreen> with TickerProviderStateMixin {
+  late final AnimationController _controller;
+  bool _messageVisible = false;
+
+  @override
+  void initState() {
+    _controller = AnimationController(
+      vsync: this,
+      duration: const Duration(milliseconds: _animationDurationInMilliseconds),
+    );
+    Future.delayed(
+        const Duration(milliseconds: _animationDurationInMilliseconds), () {
+      setState(() {
+        _messageVisible = true;
+      });
+    });
+
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: Container(
+        padding: const EdgeInsets.all(space20),
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          crossAxisAlignment: CrossAxisAlignment.center,
+          children: [
+            Lottie.network(
+              'https://assets2.lottiefiles.com/packages/lf20_pmYw5P.json',
+              controller: _controller,
+              onLoaded: (composition) {
+                _controller.duration = composition.duration;
+                _controller.forward().then(
+                      (_) => {
+                        Future.delayed(
+                          const Duration(
+                            milliseconds: _animationDurationInMilliseconds,
+                          ),
+                          () => context.pop(),
+                        ),
+                      },
+                    );
+              },
+            ),
+            const SizedBox(height: space20),
+            AnimatedOpacity(
+              duration: const Duration(
+                milliseconds: _animationDurationInMilliseconds,
+              ),
+              opacity: _messageVisible ? 1.0 : 0.0,
```

Comment:
```suggestion
            AnimatedOpacity(
              duration: const Duration(
                milliseconds: _animationDurationInMilliseconds,
              ),
              onEnd: () => context.pop(),
              opacity: _messageVisible ? 1.0 : 0.0,
```
Then we can dismiss the screen here.
By doing this, we create a sequence flow from start to end without the risk of race condition in case using `Future.delayed` incorrectly